google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java

```diff
@@ -278,6 +303,83 @@ static String getRoutingHeader(String streamName, String location) {
     return project + "locations/" + location;
   }
 
+  private Attributes buildOpenTelemetryAttributes() {
+    AttributesBuilder builder = Attributes.builder().put(telemetryKeyStreamId, this.streamName);
+    builder.put(telemetryKeyWriterId, this.writerId);
+    if ((this.traceId != null) && !this.traceId.isEmpty()) {
+      String[] traceIdParts = this.traceId.split(":", 5);
+      for (int i = 0; i < traceIdParts.length; i++) {
```

Comment:
Are we sure we want all the parts in trace id to appear as key?

Reply:
I have revised this logic as follows:

* only extract up to the first 3 colon-separated pieces of the traceId
* ignore any piece that is empty

We can consider adding more parts of the traceId based on customer feedback.

Reply:
I have now made the logic to handle traceId specific to Dataflow scenarios only.